tests/compare_request_results.py

```diff
@@ -226,61 +233,100 @@ def generate_diff(response1, response2):
     return '\n'.join([x for x in diff])
 
 
-def retrieve_response(url, request_string):
+def retrieve_response(url, request_string, desc, d: dict):
     """ POSTS request to server at url and returns response"""
+    trace()
+    start_time = datetime.datetime.now()
     request = urllib.request.Request(url,
                                      request_string.encode('utf-8'),
                                      headers={'Content-type': 'text/xml; charset=utf-8'})
     response = urllib.request.urlopen(request)
+    stop_time = datetime.datetime.now()
+    info("Time passed retriving from '" + desc + "' : " + str(stop_time - start_time))
+    d["res"] += (stop_time - start_time)
     return response.read()
 
 
 def prune_and_prettyprint(xml_string):
     """ removed nodes found in the IGNORE list and pretty print xml"""
-    parser = etree.XMLParser(remove_blank_text=True, encoding="UTF-8")
-    xml = etree.fromstring(xml_string, parser)
+    trace()
+    try:
+        parser = etree.XMLParser(remove_blank_text=True, encoding="UTF-8")
+        xml = etree.fromstring(xml_string, parser)
 
-    for path in IGNORE:
-        nodes = xml.xpath(path, namespaces=NAMESPACES)
-        for node in nodes:
-            node.getparent().remove(node)
+        for path in IGNORE:
+            nodes = xml.xpath(path, namespaces=NAMESPACES)
+            for node in nodes:
+                node.getparent().remove(node)
 
-    return etree.tostring(xml, pretty_print=True)
+        return etree.tostring(xml, pretty_print=True)
+    except Exception:
+        error("Exception while parsing response. xml_string is : \n" + xml_string.decode())
+        raise
```

Comment:
The purpose of this handler was to get it to print the argument to the function/the content it can not parse, when it has a parse error. I then rethrow the error (raise) to let the top error handler handle the exception. This is mostly for debug.